styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -122,6 +122,14 @@ public QueuedStateManager(
     this.log = Objects.requireNonNull(logger, "logger");
   }
 
+  @Override
+  public void tick() {
+    var states = getActiveStates();
```

Comment:
There will be competition with the state machine. Do we need to ensure all handlers are sort of idempotent, and we  rely on the counter to take care of duplicated events fired from the handlers?

Reply:
`TransitionLogger` might output `transition ->` for each `tick` when the workflow instance is running. It is not harmful, but could be confusing.

Reply:
Yes, all handlers should be idempotent. Should I rephrase the PR description?

Perhaps the `TransitionLogger` should also be an event consumer.

Reply:
Wrt competition, note that our schedulers already compete.

Reply:
Another comment wrt competition, I would like to work on reducing contention overall in another PR.

Reply:
> all handlers should be idempotent

We can put this into `OutputHandler` interface.

Reply:
I'm fine changing `TransitionLogger` as well, because it will be quite confusing when looking at the log.

Reply:
Done